synapse/storage/databases/main/event_federation.py

```diff
@@ -1292,6 +1292,95 @@ def _get_backfill_events(
 
         return event_id_results
 
+    @trace
+    async def record_event_failed_backfill_attempt(
```

Comment:
> (trusting you that you have useful uses for this information)
>
> *-- @reivilibre, https://github.com/matrix-org/synapse/pull/13589#pullrequestreview-1092236864*

The next PR that uses this information is ready to go: https://github.com/matrix-org/synapse/pull/13635

If this PR looks good, feel free to start reviewing that one as well. Or not and I can do the merge dance and put it in the review queue (no worries).